aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1687,6 +1687,14 @@ AT_ERROR("cholesky: MAGMA library not found in "
 }
 
 Tensor _cholesky_helper_cuda(const Tensor& self, bool upper) {
+#ifdef USE_CUSOLVER
+  return _cholesky_helper_cuda_lib(self, upper);
+#else
+  return _cholesky_helper_cuda_legacy(self, upper);
```

Comment:
I think naming this with `_cusolver` and `_magma` at the end is more descriptive.

Reply:
I totally support this idea.

Cholesky `potrf` and `potrfBatched` are both under cusolver API. However, matrix inverse `getrf` is cusolver, but `getrfBatched` is cublas. So using `_cusolver` for all functions might be tricky.

Reply:
I think `getrfBatched` should not have been placed together with a for loop of `getrf` under the same function `apply_batched_inverse_lib`. The reason is exactly that one is in cuSOLVER and the other is in cuBLAS. `apply_batched_inverse_lib` is guarded under `USE_CUSOLVER`, but `getrfBatched` is available even for `USE_CUSOLVER=0`, so in principle I think it would be better to have it separate and be able to use `getrfBatched` path if `USE_CUSOLVER=0`. But that's a different discussion.

We could use `cholesky_helper_magma`, `cholesky_helper_cusolver`, `cholesky_helper_batched_cusolver` here and similarly for other new functions. For inverse it would be `inverse_helper_magma`, `inverse_helper_cusolver`, `inverse_helper_batched_cublas`. Alternatively using namespaces `at::native::cuda::magma::*`, `at::native::cuda::cusolver::*`, `at::native::cuda::cublas::*`.